src/hu_nmt/data_augmentator/graph_mappers/edge_mapper.py

```diff
@@ -0,0 +1,161 @@
+import networkx as nx
+from collections import defaultdict, Counter
+import nltk
+from typing import List
+
+from hu_nmt.data_augmentator.graph_mappers.graph_similarity_base import GraphSimilarityBase, Edge, Node
+
+
+class EdgeMapper(GraphSimilarityBase):
+    def __init__(self):
+        self.dep_weights = defaultdict(lambda: 1)
+
+        self.dep_weights['nsubj'] = 3
+        self.dep_weights['obj'] = 3
+        self.dep_weights['nmod'] = 2
+        self.dep_weights['obl'] = 2
+        self.dep_weights['case'] = 2
+        self.dep_weights['amod'] = 2
+        self.dep_weights['advmod'] = 2
+
+    def map_edges(self, g1: nx.DiGraph, g2: nx.DiGraph):
+        g1 = self.add_weight(g1)
+        g2 = self.add_weight(g2)
+        mapping = {}
+
+        g1_edges = list(sorted(g1.edges(data=True), key=lambda x: -x[2]['weight']))
+        g2_edges = list(sorted(g2.edges(data=True), key=lambda x: -x[2]['weight']))
+
+        for i, (s1, d1, data1) in enumerate(g1_edges):
+            cands = list(filter(lambda x: x[2]['dep'] == data1['dep'], g2_edges))
+            if len(cands) == 1:
+                mapping[(s1, d1)] = (cands[0][0], cands[0][1])
+                g2_edges.remove(cands[0])
+            elif len(cands) > 1:
+                max_cands = self.get_max_cands((s1, d1, data1), cands, g1, g2)
+                if len(max_cands) == 1:
+                    mapping[(s1, d1)] = (max_cands[0][0], max_cands[0][1])
+                    g2_edges.remove(max_cands[0])
+                else:
+                    min_routes = self._get_max_cand_by_route((s1, d1, data1), max_cands, g1, g2)
+                    if len(min_routes) == 1:
+                        mapping[(s1, d1)] = (min_routes[0][0], min_routes[0][1])
+                        g2_edges.remove(min_routes[0])
+                    else:
+                        max_children = self._get_max_children((s1, d1, data1), max_cands, g1, g2)
+                        mapping[(s1, d1)] = (max_children[0][0], max_children[0][1])
+                        g2_edges.remove(max_children[0])
+        return mapping
```

Comment:
Can you please add comments to this function (and the more important ones below) explaining the steps? It's kinda hard to follow without knowing the maths.